src/scipp/html/table.py

```diff
@@ -0,0 +1,230 @@
+                                  indices=indices,
+                                  need_bin_edge=bin_edges,
+                                  is_bin_edge=da.attrs.is_edges(name),
+                                  border=False))
+
+    return out
+
+
+def _make_entries_header(ds: Dataset) -> str:
+    out = '<tr>'
+    if ds.coords:
+        out += f'<th colspan="{len(ds.coords)}"></th>'
+    for name, da in sorted(ds.items()):
+        ncols = 1 + len(da.masks) + len(da.attrs)
+        out += f'<th {CENTER} colspan="{ncols}">{name}</th>'
+    out += '</tr>'
+    return out
+
+
+def _make_sections_header(ds: Dataset) -> str:
+    out = '<tr>'
+    if ds.coords:
+        out += f'<th {CENTER} colspan="{len(ds.coords)}">Coordinates</th>'
+    for _, da in sorted(ds.items()):
+        out += f'<th {CENTER_BORDER}>Data</th>'
+        if da.masks:
+            out += f'<th {CENTER} colspan="{len(da.masks)}">Masks</th>'
+        if da.attrs:
+            out += f'<th {CENTER} colspan="{len(da.attrs)}">Attributes</th>'
+    out += '</tr>'
+    return out
+
+
+def _to_html_table(header: str, body: List[list]) -> str:
+    out = '<table>' + header
+    ncols = len(body)
+    nrows = len(body[0])
+    for i in range(nrows):
+        out += '<tr>' + ''.join([body[j][i] for j in range(ncols)]) + '</tr>'
+    out += '</table>'
+    return out
+
+
+def _find_bin_edges(ds: Dataset) -> bool:
+    for key in ds.coords:
+        if ds.coords.is_edges(key):
+            return True
+    for da in ds.values():
+        for key in da.attrs:
+            if da.attrs.is_edges(key):
+                return True
+    return False
+
+
+def _strip_scalars_and_broadcast_masks(ds: Dataset) -> Dataset:
+    out = Dataset()
+    for key, da in ds.items():
+        if da.ndim == 1:
+            out[key] = DataArray(data=da.data,
+                                 coords={
+                                     key: var
+                                     for key, var in da.coords.items()
+                                     if var.dims == da.data.dims
+                                 },
+                                 attrs={
+                                     key: var
+                                     for key, var in da.attrs.items()
+                                     if var.dims == da.data.dims
+                                 },
+                                 masks={
+                                     key: var.broadcast(sizes=da.sizes)
+                                     for key, var in da.masks.items()
+                                 })
+    return out
+
+
+def _to_dataset(obj: Union[VariableLike, dict]) -> Dataset:
+    if isinstance(obj, DataArray):
+        return Dataset({obj.name: obj})
+    if isinstance(obj, Variable):
+        return Dataset(data={"": obj})
+    if isinstance(obj, dict):
+        return Dataset(obj)
+    return obj
+
+
+def table(obj: Union[VariableLike, dict], max_rows: Optional[int] = 20):
```

Comment:
```suggestion
def table(obj: Union[VariableLike, Dict[str, Union[Variable, DataArray]]], max_rows: int = 20):
```

Reply:
:+1: for the Dict, why remove the `Optional`? Should I remove `Optional` in other places as well?

Reply:
`Optional[T]` is basically an alias for `Union[T, None]`. So if `table` does not support `max_rows=None`, it can't be `Optional`. (This has nothing to do with whether there is a default value.)

Reply:
> This has nothing to do with whether there is a default value.

Ok I wasn't aware of that subtlety.